libs/full/runtime_distributed/src/server/runtime_support_server.cpp

```diff
@@ -227,8 +227,11 @@ namespace hpx { namespace components { namespace server {
     void runtime_support::dijkstra_make_black()
     {
         // Rule 1: A machine sending a message makes itself black.
-        std::lock_guard<dijkstra_mtx_type> l(dijkstra_mtx_);
-        dijkstra_color_ = true;
+        if (!dijkstra_color_)
```

Comment:
Wouldn't accessing the variable `dijkstra_color_` outside a lock create a race? Granted, the race may be benign -- nevertheless. Should we convert that variable into an atomic instead?